### PR TITLE
Improve instructions for new developers

### PR DESCRIPTION
**Pull request by mrsimpson**
## 📚 Purpose

As a new developer, it should be as easy as possible to start from scratch and get things up and running to evaluate the project.
This PR is the result of my findings as I tried to locally build, launch and hopefully utilize the combined micro services on a local (Mac) k8s.

## 👌 Resolves:
- [X] necessary dependencies to build the images locally
- [X] Some basic troubleshooting when launching the services on a local cluster

## 📦 Impacts:
Only documentation

## ✅ PR Checklist

Some parts are missing. I'd love to get clarification about it
- [x] Information about necessary configuration (`mds-config-files`, `cloud-config/env`): What's in those files, is there a template?
- Istio-related questions have been extracted to #899 and might result in a new PR

### CODE REVIEW

**Comment by mrsimpson** [comment]
Thanks, @avatarneil !
Currently we are using keycloak as oauth provider. Knowing that issuance can be external, I'll be looking into how to set this up.

File: README.md
Comment (avatarneil) [reply README.md:277]: Once #892 lands, we should be able to just delete this section I think :)

Comment (avatarneil) [reply README.md:277]: @mrsimpson I think this can be removed now that #892 merged!

**Review by avatarneil — APPROVED** [review]

File: README.md
Comment (PeterSP) [inline README.md:281]: How about using the next level of heading?
```suggestion
#####`tl;dr`
```

File: README.md
Comment (PeterSP) [inline README.md:291]:
```suggestion
##### If image pull fails
```

File: README.md
Comment (PeterSP) [inline README.md:295]:
```suggestion
##### If scheduling fails due to limited resources
```

**Review by PeterSP — COMMENTED** [review]
Minor nits, otherwise looks great!!

**Comment by mrsimpson** [comment]
@PeterSP Thanks for the nit picking (no sarcasm!). I much appreciate passion for quality work (no matter the technology)!
I opened #896 in order to facilitate this process (based upon this branch)

**Review by nessur — APPROVED** [review]

**Comment by mrsimpson** [comment]
As I consider this PR now "complete with everything not related to Istio", I wondered whether I should better move this part to a discussion (and get this one merged)? I could subsequently open up a new PR once the Istio-stuff can be added. WDYT, especially @avatarneil  and @invertigo ?

**Comment by avatarneil** [comment]
@mrsimpson That sounds like a plan to me, if you've got something ready to merge, I'd merge it!

**Comment by mrsimpson** [comment]
I opened #899 . How's the usual procedure in your software development workflow? Who merges PRs and when?

**Comment by mrsimpson** [comment]
Bildschirmfoto 2022-02-15 um 22 42 49

**Comment by avatarneil** [comment]
Please forgive the PR spam.. working on getting Jenkins to run builds for outside-of-org collaborators.

**Comment by avatarneil** [comment]
Please Jenkins ;-;

**Comment by mrsimpson** [comment]
As per his waiteral appearance, I'd expect him to be more customer-focused – particularly as you, @avatarneil, have been very polite to Mr. Jenkins. 
We moved to Gitlab-CI. Does feature a comparatively nice appearance, but well, works like a workhorse...

Coming back to the PR: I can see auto-merge enabled.

**Comment by avatarneil** [comment]
It seems like right now we have restrictions on what branches Jenkins will run tests for (it does not run for PRs coming form forks). While we wait for feedback from our devops/security team on how we can safely run Jenkins against PRs from forks, I'll just force-merge bypassing Jenkins (on any PRs which actually touch the code, I'll run tests on my machine, too).

**Pull request merged**
